### PR TITLE
add sanitize_string function

### PR DESCRIPTION
**Pull request by sorelfitzgibbon**
- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have reviewed the [Nextflow pipeline standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3193890/Nextflow+pipeline+standardization).

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using \[AD_username (or 5 letters of AD if AD is too long)]-\[brief_description_of_branch].

- [x] I have set up or verified the branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] I have added my name to the contributors listings in the
``metadata.yaml`` and the ``manifest`` block in the `nextflow.config` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

- [x] I have added the changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

- [x] I have updated the version number in the `metadata.yaml` and `manifest` block of the `nextflow.config` file following [semver](https://semver.org/), or the version number has already been updated. (*Leave it unchecked if you are unsure about new version number and discuss it with the infrastructure team in this PR.*)

- [x] I have tested the config being added or modified. Outline the tests below.

Added the `sanitize_string` function to `common_methods`.  

### Testing
      
 Tested in `pipeline-SQC-DNA` with local function removed (branch `sfitz-use-sanitize-from-common`)
output: `/hot/software/pipeline/pipeline-SQC-DNA/Nextflow/development/unreleased/sfitz-use-sanitize-from-common/a_mini_n2-std-input-default`
log: `/hot/software/pipeline/pipeline-SQC-DNA/Nextflow/development/unreleased/sfitz-use-sanitize-from-common/slurm-85433.out`

### CODE REVIEW

**Comment by yashpatel6** [comment]
Generally looks good! Can you also update the README in config/methods to indicate the added function and update the CHANGELOG?

**Review by zhuchcn — COMMENTED** [review]
This function seems to be very specific and should be only used to sanitize our internal patient/sample IDs, rather than sanitizing any string. So maybe we should consider renaming it to be more specific, like `sanitize_cds_ids`, or putting it into a separate namespace together with other functions that handle related jobs. Where will this function used?

File: config/methods/README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ To use a common `methods` function:
 - `set_env` - Function to set the workDir depending on parameters and on Slurm job ID
 - `setup_docker_cpus` - Function to use allocated CPUs for a process to specify number of CPUs rather than CPU shares with Docker
 - `get_absolute_path` - Function to resolve a path relative to the currently-loading configuration file into an absolute path
+- `sanitize_uclahs_cds_id` - Pass input string to sanitize, i.e. keeping only alphanumeric, `-`, `/`, and `.` characters and replacing `_` with `-`
```
Comment (yashpatel6) [inline config/methods/README.md:41]: I'd suggest throwing in a quick example in the list of examples below for this function

**Review by yashpatel6 — COMMENTED** [review]
One minor suggestion for the documentation, anything else to add @zhuchcn ?

**Review by zhuchcn — APPROVED** [review]
Looks good!

**Review by yashpatel6 — APPROVED** [review]

**Pull request merged**